streams/src/main/java/org/apache/kafka/streams/state/VersionedRecord.java

```diff
@@ -27,6 +27,7 @@
 public final class VersionedRecord<V> {
     private final V value;
     private final long timestamp;
+    private final long validTo;
```

Comment:
Btw: given that `VersionedKeyQuery` also returns a `VersionedRecord`, I believe we need to update it to set this timestamp correctly for historic queried? If we query "latest" there is nothing to be done, but if we query an older version based on an `asOfTimestamp` we might need to take care of this case?

Might also be worth mentioning this on the KIP.